### PR TITLE
skip some lightning tests if falling back to default.qubit

### PR DESCRIPTION
**Pull request by timmysilv**
CI fails occasionally because these tests expect lightning-specific failures, but lightning itself didn't install. It might be better to do some larger skip if we're falling back on default.qubit, but this will at least stop intermittent failures when lightning fails to build.

PS: should we not inspect this private attribute of the device? I figured it made sense, but we could do an `isinstance(dev, DefaultQubit)` check instead

### CODE REVIEW

**Review by mudit2812 — APPROVED** [review]
One of the approving reviews should probably be someone from the performance squad. But looks good to me :)

**Comment by mlxd** [comment]
Hi @timmysilv 
While I understand the want here, I am not in favour of permitting the CI to pass through if it is in an inconsistent state (i.e. lightning wasn't properly fetched from the TestPyPI).

In this case, I will approve these changes, but request that a new test is added to validate a working lightning build is successfully installed. that way, a manual override can be allowed once this is known to be the problem. How does that sound?

**Review by mlxd — CHANGES_REQUESTED** [review]
Blocked on above comment.

**Comment by timmysilv** [comment]
[here](https://github.com/PennyLaneAI/pennylane/actions/runs/5015260285/jobs/8990585212)'s an example failing run. @mlxd I'm not sure what exactly that test would look like, since the issue here is that lightning seems to fail intermittently. I guess I have a question for you, which is - do you know why it only fails intermittently? If we can instead fix that, I'd be happiest because I'd definitely rather do none of this. If we can't test the actual lightning, but we need lightning tests to pass for the PR to be merge-ready, then I don't want to skip these tests at all. I want to fix the root of the problem.

I made this PR because I figured we don't know why, and I wanted to unblock developers who might be confused by the failure. This can probably be handled better.

**Comment by mlxd** [comment]
> [here](https://github.com/PennyLaneAI/pennylane/actions/runs/5015260285/jobs/8990585212)'s an example failing run. @mlxd I'm not sure what exactly that test would look like, since the issue here is that lightning seems to fail intermittently. I guess I have a question for you, which is - do you know why it only fails intermittently? If we can instead fix that, I'd be happiest because I'd definitely rather do none of this. If we can't test the actual lightning, but we need lightning tests to pass for the PR to be merge-ready, then I don't want to skip these tests at all. I want to fix the root of the problem.
> 
> I made this PR because I figured we don't know why, and I wanted to unblock developers who might be confused by the failure. This can probably be handled better.

The failure is likely due to an in-progress lightning development release update to TestPyPI. whenever we merge to master, wheels are built to upload there so that they can be reused by other CI instances, such as for PennyLane, the QML repo, and the non-CPU lightning tests. The first wheel to upload is always the non-binary wheel, as it is trivial to build. If the CI hits that before the compiled wheels are uploaded, which can take between 5-15 mins longer, it will be pulled down and cause what you observe. I think the quickest fix here is to simply avoid the non-binary wheel to be uploaded to TestPyPI which will always guarantee an arch specific version is available to be pulled, even if the arch you want is not yet uploaded, the previous one will be.

How does that sound?

**Comment by mlxd** [comment]
If you want to make a quick PR to remove the || statement second term here https://github.com/PennyLaneAI/pennylane-lightning/blob/008664b2dfd9e17f4979b43707d71be6bd1b0486/.github/workflows/wheel_noarch.yml#LL59C44-L59C44
that should fix it with the next lightning merge to master

**Comment by timmysilv** [comment]
the above linked PR should suffice

**Pull request closed without merging**
